components/core/src/clp_s/JsonFileIterator.cpp

```
@@ -64,6 +71,93 @@ bool JsonFileIterator::read_new_json() {
+                std::string char_details;
+                for (auto const& [ch, count] : result.sanitized_char_counts) {
+                    if (!char_details.empty()) {
+                        char_details += ", ";
+                    }
+                    char_details
+                            += fmt::format("0x{:02x} ({})", static_cast<unsigned char>(ch), count);
+                    total_sanitized += count;
+                }
+                size_t bytes_added = m_buf_occupied - old_buf_occupied;
+                SPDLOG_WARN(
+                        "Escaped {} control character(s) in JSON{}: [{}]. Buffer expanded by {} "
```

Cppcheck (2.19.0)

[style] The function 'read_numeric_value' is never used.

(unusedFunction)